### PR TITLE
Add unit tests for agentclusterinstall

### PR DESCRIPTION
**Pull request by trewest**
Adds unit tests for the agentclusterinstall resource in the assisted pkg

Needed for https://github.com/openshift-kni/eco-goinfra/issues/295

### CODE REVIEW

File: pkg/assisted/agentclusterinstall_test.go [hunk 1]
```diff
@@ -0,0 +1,1180 @@
...
+		{
+			count:         1,
+			expectedError: "",
+		},
+		{
+			count:         0,
+			expectedError: "agentclusterinstall controlplane agents must be greater than 0",
+		},
+		{
+			count:         -1,
```
Comment (achuzhoy) [inline pkg/assisted/agentclusterinstall_test.go:606]: isn't this one redundant?

File: pkg/assisted/agentclusterinstall_test.go [hunk 2]
```diff
@@ -0,0 +1,1180 @@
...
+
+func TestWithImageSet(t *testing.T) {
+	testCases := []struct {
+		image         string
+		expectedError string
+	}{
+		{
+			image:         "4.16",
+			expectedError: "",
+		},
```
Comment (achuzhoy) [inline pkg/assisted/agentclusterinstall_test.go:628]: One more test with something that raises an error?

File: pkg/assisted/agentclusterinstall_test.go [hunk 3]
```diff
@@ -0,0 +1,1180 @@
...
+		{
+			count:         1,
+			expectedError: "",
+		},
+		{
+			count:         0,
+			expectedError: "",
+		},
+		{
```
Comment (achuzhoy) [inline pkg/assisted/agentclusterinstall_test.go:657]: seems redundant

**Review by achuzhoy — APPROVED** [review]

File: pkg/assisted/agentclusterinstall_test.go (hunk 1 above)
Comment (trewest) [reply pkg/assisted/agentclusterinstall_test.go:606]: 0 and -1 are used to catch edged cases. If WithControlPlaneAgents has it's validation for agentCount removed or changed to allow 0, these tests will catch it.

File: pkg/assisted/agentclusterinstall_test.go (hunk 2 above)
Comment (trewest) [reply pkg/assisted/agentclusterinstall_test.go:628]: Currently this method doesn't return any errors. I will see if there is a validation that can be added before. It was allowed to be empty in the past

File: pkg/assisted/agentclusterinstall_test.go (hunk 3 above)
Comment (trewest) [reply pkg/assisted/agentclusterinstall_test.go:657]: Which part?

File: pkg/assisted/agentclusterinstall_test.go (hunk 2 above)
Comment (trewest) [reply pkg/assisted/agentclusterinstall_test.go:628]: No, it looks like an empty imageset reference can still be passed in

File: pkg/assisted/agentclusterinstall_test.go [hunk 4]
```diff
@@ -0,0 +1,1179 @@
...
+				builder.Definition.Spec.HoldInstallation = true
+
+				return builder, nil
+			},
+			validator: func(acib *AgentClusterInstallBuilder) bool {
+				return acib.Definition.Spec.HoldInstallation == true
+			},
+		},
+		{
+			option: func(builder *AgentClusterInstallBuilder) (*AgentClusterInstallBuilder, error) {
```
Comment (kononovn) [inline pkg/assisted/agentclusterinstall_test.go:402]: Add option which returns simple error.

File: pkg/assisted/agentclusterinstall_test.go [hunk 5]
```diff
@@ -0,0 +1,1179 @@
...
+
+		_, err := testCase.option(testBuilder)
+
+		assert.Nil(t, err)
+
+		assert.True(t, testCase.validator(testBuilder))
+	}
+}
+
+func TestWithApiVip(t *testing.T) {
```
Comment (kononovn) [inline pkg/assisted/agentclusterinstall_test.go:427]: Please rename all functions below to:
TestAgentClusterInstallWith*

**Review by kononovn — CHANGES_REQUESTED** [review]
2 minor comments

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**